### ISSUE TITLE
Return values of scopes with `whereDate`

### ISSUE THREAD

**Issue by brendt** [nunomaduro/larastan#89]
I've got this method in a model

```php
use Illuminate\Database\Eloquent\Builder;

public function scopeBeforeFuture(Builder $builder): Builder
{
    return $builder->whereDate('date_from', '<=', now()->toDateString());
}
```

And get this error

```
Method App\Domain\Unit\Models\UnitTypePrice::scopeBeforeFuture() should return Illuminate\Database\Eloquent\Builder but returns Illuminate\Database\Query\Builder.
```

I believe this has to do with `whereDate` having this return type `@return \Illuminate\Database\Query\Builder|static`. ?

**Comment by nunomaduro** [comment]
@brendt This issue got addressed on the version `v0.2.11`. Can you test it?

**Comment by brendt** [comment]
It's fixed, thanks!
